src/elements/content-sidebar/Sidebar.js

```diff
@@ -122,6 +128,44 @@ class Sidebar extends React.Component<Props, State> {
         }
     }
 
+    getIsOpen = () => {
+        const { isDefaultOpen } = this.props;
+        return this.isForcedSet() ? this.isForcedOpen() : !!isDefaultOpen;
+    };
+
+    addAnnotation() {
+        const {
+            annotatorState: { action, annotation, meta },
+            api,
+            currentUser,
+            file,
+            getAnnotationsMatchPath,
+            history,
+        } = this.props;
+        const { requestId } = meta || {};
+        const isOpen = this.getIsOpen();
```

Comment:
Should all of this code live in SidebarPanels, instead, where `isOpen` is a simple prop? Or maybe even as its own HOC around SidebarPanels?